rust-lang/cargo
# refactor(job_queue): docs and move types around

### What does this PR try to resolve?

Refine module-level documentation for `job_queue`, and trying to move types to proper places.

- `job_queue.rs` -> `job_queue/mod.rs`
- `job.rs` -> `job_queue/job.rs`
- `JobState` -> `job_queue/job_state.rs`

### How should we test and review this PR?

Run `cargo doc --document-private-items --no-deps --open` to proofread the doc.

This shouldn't affect anything. Only one thing need to make sure it doesn't go wrong. The major logic of [`doit`](https://github.com/rust-lang/cargo/blob/7ddcf0fe3348b7e0f4ad4a730eab60a20638ef28/src/cargo/core/compiler/job_queue.rs#L1122-L1168) closure is moved into `JobState::run_to_finish`. It shouldn't change the behavior, though.


### Additional information

Does anyone think it beneficial to have a graph explaining how job queue works? Something like this (from `puma/puma`)

![https://bit.ly/2zwzhEK](https://github.com/puma/puma/raw/master/docs/images/puma-connection-flow.png)
